chain/lf3/f3.go

```diff
@@ -0,0 +1,164 @@
+package lf3
+
+import (
+	"context"
+	"errors"
+	"time"
+
+	"github.com/ipfs/go-datastore"
+	"github.com/ipfs/go-datastore/namespace"
+	logging "github.com/ipfs/go-log/v2"
+	pubsub "github.com/libp2p/go-libp2p-pubsub"
+	"github.com/libp2p/go-libp2p/core/host"
+	"go.uber.org/fx"
+	"golang.org/x/xerrors"
+
+	"github.com/filecoin-project/go-f3"
+	"github.com/filecoin-project/go-f3/blssig"
+	"github.com/filecoin-project/go-f3/certs"
+	"github.com/filecoin-project/go-f3/gpbft"
+
+	"github.com/filecoin-project/lotus/api"
+	"github.com/filecoin-project/lotus/build"
+	"github.com/filecoin-project/lotus/chain/stmgr"
+	"github.com/filecoin-project/lotus/chain/store"
+	"github.com/filecoin-project/lotus/node/modules/dtypes"
+	"github.com/filecoin-project/lotus/node/modules/helpers"
+)
+
+type F3 struct {
+	inner *f3.F3
+
+	signer gpbft.Signer
+}
+
+type F3Params struct {
+	fx.In
+
+	NetworkName  dtypes.NetworkName
+	PubSub       *pubsub.PubSub
+	Host         host.Host
+	ChainStore   *store.ChainStore
+	StateManager *stmgr.StateManager
+	Datastore    dtypes.MetadataDS
+	Wallet       api.Wallet
+}
+
+var log = logging.Logger("f3")
+
+func New(mctx helpers.MetricsCtx, lc fx.Lifecycle, params F3Params) (*F3, error) {
+	manifest := f3.LocalnetManifest()
+	manifest.NetworkName = gpbft.NetworkName(params.NetworkName)
+	manifest.ECDelay = 2 * time.Duration(build.BlockDelaySecs) * time.Second
```

Comment:
So if `EcDelay` and `EcPeriod` are always the same, why do they both exist in `manifest` ? I assume this will change in the future ?

Reply:
I believe it will. Cc @Kubuxu

Reply:
(IMO, ECDelay should just be a multiple of ECPeriod, but yeah, it'll probably change)